### ISSUE TITLE
Proposal: Enhance KDropdownMenu for Context Menu Support

### ISSUE THREAD

**Issue by AlexVelezLl** [learningequality/kolibri-design-system#571]
## Product
For now there are some examples in Studio, but it can be useful for Kolibri and other products aswell.

## Current behavior

The current implementation of the KDropdownMenu doesnt support being rendered as a context menu as it doesnt allow passing the exact position where it should be rendered (something that is required for rendering in the position of the right click).

Also there is no a reusable solution within KDS to support showing a menu on right click, in studio we have [this component](https://github.com/learningequality/studio/blob/d6e919c56492687258eba63a923a9c0d32c1e174/contentcuration/contentcuration/frontend/shared/views/ContextMenuCloak.vue#L1), but it requires to have a vuex module to save a general state and to know which menu are we showing, as we should have only one context menu rendered at a time. 

## Desired behavior

Enable the KDropdownMenu component to function as a context menu, allowing users to trigger it via right-click interactions.

For this, one option could be keeping the current implementation of the `KDropdownMenu` but support a `positionX` and `positionY` relative to the parent element as props. This position can then be passed as an [offset](https://atomiks.github.io/tippyjs/v6/all-props/#offset) property to tipty, [which is the library that is used in UiPopover](https://github.com/learningequality/kolibri-design-system/blob/7ae0990e0864397718272543baf6d68b79430fbe/lib/keen/UiPopover.vue#L169) to place the menu in the DOM.

To provide the `positionX` and `positionY` we can create a new composable that has a global variable about which context menu is being presented, and that registers the handler of the right click event on the element or on the element's parent of the component that calls the composable, with this we could probably have a composable that returns `{ isActive, positionX, positionY }`.

With this we could have something like (having `useContextMenu` attaching the behaviour to `this.$el`):

```vue
<template>
  ...
  <KDropdownMenu
    v-if="isActive"
    ...
   :positionX="positionX"
   :positionY="positionY"
  />
  ...
</template>

<script>
export default {
  ...,
  setup() {
    const { isActive, positionX, positionY } = useContextMenu();
    return { isActive, positionX, positionY }; 
  }
}
<script>
```

A more clean but coupled solution could be that the `KDropdownMenu` accepts a `contextMenu` prop and that it internally uses the `useContextMenu` and attaches the right click behavior to the immediate parent of the KDropdownMenu in the DOM, and we could end with a more dynamic component:

```vue
<template>
  ...
  <div class="card">
    ...
    <KDropdownMenu
      <!-- Any right click on the card component will active the context menu -->
      contextMenu
      :options="options"
      ...
    />
  </div>
</template>
```

The main benefit of this last solution is that if we need to use the same menu attached to a button, or as a context menu, the difference would only lie in changing a prop, as for example in the case of the [ContentNodeOptions](https://github.com/learningequality/studio/blob/d6e919c56492687258eba63a923a9c0d32c1e174/contentcuration/contentcuration/frontend/channelEdit/components/ContentNodeOptions.vue#L1) component in studio. And also, as it is not attached to `this.$el` of the caller component, we wouldnt need to break down components if we need to attach a the behaviour on an element other than the root element of the caller component.

**Comment by AlexVelezLl** [comment]
I created a follow up issue to reconsider the dropdown menus interface in #634.

**Comment by AlexVelezLl** [comment]
Closed by #583
